src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -526,6 +525,49 @@ await invokeTask.NoThrowAwaitable(); // may or may not throw due to cancellation
             });
     }
 
+    [Fact]
+    public void InvokeAsync_ProcessParameterSentAsArray()
+    {
+        var sending = new MemoryStream();
+
+        var rpc = JsonRpc.Attach(sendingStream: sending, receivingStream: null, target: null, passParameterAsObject: false);
+        rpc.NotifyAsync("test", new Foo() { Bar = "test" }).Wait();
+
+        sending.Position = 0;
+        var handler = new HeaderDelimitedMessageHandler(null, sending);
+        var json = handler.ReadAsync(default(CancellationToken)).GetAwaiter().GetResult();
+
+        var message = JsonRpcMessage.FromJson(json, new JsonSerializerSettings());
+        Assert.Equal(message.Parameters.Type, JTokenType.Array);
+    }
+
+    [Fact]
+    public void InvokeAsync_ProcessParameterSentAsObject()
+    {
+        var sending = new MemoryStream();
+
+        var rpc = JsonRpc.Attach(sendingStream: sending, receivingStream: null, target: null, passParameterAsObject: true);
+        rpc.NotifyAsync("test", new Foo() { Bar = "test" }).Wait();
```

Comment:
Can you please add a test that uses an anonymous type? (e.g. `new { foo = 3, bar = "3" }`)